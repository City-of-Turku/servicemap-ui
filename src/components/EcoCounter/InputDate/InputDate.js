import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';
import { InputBase, InputAdornment } from '@mui/material';
import styled from '@emotion/styled';
import { CalendarMonth } from '@mui/icons-material';

const InputDate = ({ value, onClick, onChange }, ref) => (
  <StyledInputBase
    type="text"
    value={value}
    ref={ref}
    onChange={e => onChange(e.target.value)}
    onClick={onClick}
    startAdornment={(
      <InputAdornment position="start">
        <CalendarMonth />
      </InputAdornment>
    )}
  />
);

const StyledInputBase = styled(InputBase)(() => ({
  fontSize: '0.913rem',
  fontWeight: 'bold',
  boxSizing: 'border-box',
  backgroundColor: '#fff',
  cursor: 'pointer',
}));

InputDate.propTypes = {
  value: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
};

InputDate.defaultProps = {
  value: '',
};

export default forwardRef(InputDate);
