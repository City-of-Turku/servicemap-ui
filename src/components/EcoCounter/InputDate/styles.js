const styles = (theme) => ({
  input: {
    border: 'none',
    textAlign: 'end',
    paddingRight: theme.spacing(0.7),
    ...theme.typography.body2,
    '&:hover': {
      cursor: 'pointer',
      border: 'none',
    },
  },
});

export default styles;
