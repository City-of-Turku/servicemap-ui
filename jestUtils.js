import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });