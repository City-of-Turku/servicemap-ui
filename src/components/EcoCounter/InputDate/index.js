import { withStyles } from '@mui/styles';
import InputDate from './InputDate';
import styles from './styles';

export default withStyles(styles)(InputDate);
