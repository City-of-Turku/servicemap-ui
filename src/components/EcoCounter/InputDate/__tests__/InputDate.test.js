import React from 'react';
import { fireEvent } from '@testing-library/react';
import { getRenderWithProviders } from '../../../../../jestUtils';
import InputDate from '../index';

const mockProps = {
  value: 'Test value',
};

const renderWithProviders = getRenderWithProviders({});

describe('<InputDate />', () => {
  it('renders without crashing', () => {
    const { container } = renderWithProviders(<InputDate onClick={() => {}} onChange={() => {}} />);
    expect(container).toBeTruthy();
  });

  it('should match snapshot', () => {
    const { container } = renderWithProviders(<InputDate onClick={() => {}} onChange={() => {}} />);
    expect(container).toMatchSnapshot();
  });

  it('displays the provided value', () => {
    const { getByDisplayValue } = renderWithProviders(<InputDate value={mockProps.value} onClick={() => {}} onChange={() => {}} />);
    expect(getByDisplayValue(mockProps.value)).toBeInTheDocument();
  });

  it('calls the onClick function when clicked', () => {
    const onClickMock = jest.fn();
    const { getByRole } = renderWithProviders(<InputDate onClick={onClickMock} onChange={() => {}} />);
    fireEvent.click(getByRole('button'));
    expect(onClickMock).toHaveBeenCalled();
  });
});
