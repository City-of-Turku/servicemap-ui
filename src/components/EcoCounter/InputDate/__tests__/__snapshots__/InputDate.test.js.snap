// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputDate /> should match snapshot 1`] = `
<div>
  <div
    class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedStart css-ext27a-MuiInputBase-root"
  >
    <div
      class="MuiInputAdornment-root MuiInputAdornment-positionStart css-ittuaa-MuiInputAdornment-root"
    >
      <span
        class="notranslate"
      >
        â€‹
      </span>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-6yqi1c-MuiSvgIcon-root"
        data-testid="CalendarMonthIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 4h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2m0 16H5V10h14zM9 14H7v-2h2zm4 0h-2v-2h2zm4 0h-2v-2h2zm-8 4H7v-2h2zm4 0h-2v-2h2zm4 0h-2v-2h2z"
        />
      </svg>
    </div>
    <input
      class="MuiInputBase-input MuiInputBase-inputAdornedStart css-ngazp5-MuiInputBase-input"
      type="button"
      value=""
    />
  </div>
</div>
`;
